controllers/provisioning_controller.go

```diff
@@ -335,6 +317,68 @@ func (r *ProvisioningReconciler) provisioningInfo(provConfig *metal3iov1alpha1.P
 	}
 }
 
+//Ensure Finalizer is present on the Provisioning CR when not deleted and
+//delete resources and remove Finalizer when it is
+func (r *ProvisioningReconciler) checkForCRDeletion(info *provisioning.ProvisioningInfo) (bool, error) {
+	// examine DeletionTimestamp to determine if object is under deletion
+	if info.ProvConfig.ObjectMeta.DeletionTimestamp.IsZero() {
+		// The object is not being deleted, so if it does not have our finalizer,
+		// then lets add the finalizer and update the object. This is equivalent
+		// to registering our finalizer.
+		if !containsString(info.ProvConfig.ObjectMeta.Finalizers,
+			metal3iov1alpha1.ProvisioningFinalizer) {
+			// Add finalizer becasue it doesn't already exist
+			info.ProvConfig.ObjectMeta.Finalizers = append(info.ProvConfig.ObjectMeta.Finalizers,
+				metal3iov1alpha1.ProvisioningFinalizer)
+			if err := r.Client.Update(context.Background(), info.ProvConfig); err != nil {
+				return false, errors.Wrap(err, "failed to update Provisioning CR with its finalizer")
+			}
+		}
+		return false, nil
+	} else {
+		// The Provisioning object is being deleted
+		if containsString(info.ProvConfig.ObjectMeta.Finalizers, metal3iov1alpha1.ProvisioningFinalizer) {
+			err := r.deleteMetal3Resources(info)
+			if err != nil {
+				return false, errors.Wrap(err, "failed to delete metal3 resource")
+			}
+			// Remove our finalizer from the list and update it.
+			info.ProvConfig.ObjectMeta.Finalizers = removeString(info.ProvConfig.ObjectMeta.Finalizers,
+				metal3iov1alpha1.ProvisioningFinalizer)
+			if err = r.Client.Update(context.Background(), info.ProvConfig); err != nil {
+				return true, errors.Wrap(err, "failed to remove finalizer from Provisioning CR")
+			}
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+//Delete Secrets and the Metal3 Deployment objects
+func (r *ProvisioningReconciler) deleteMetal3Resources(info *provisioning.ProvisioningInfo) error {
+	failedSecrets := provisioning.DeleteAllSecrets(info)
```

Comment:
i think this should return an error

Reply:
Do you think deleteMetal3Resources() should return error when DeleteAllSecrets() returns error(s) or just log it and proceed with deleting other resources as it is doing now.